snakemake/executors/azure_batch.py

```diff
@@ -0,0 +1,518 @@
+__author__ = "Johannes Köster, Andreas Wilm, Jake VanCampen"
+__copyright__ = "Copyright 2022, Johannes Köster"
+__email__ = "johannes.koester@uni-due.de"
+__license__ = "MIT"
+
+import os
+from collections import namedtuple
+from urllib.parse import urlparse
+import datetime
+import uuid
+import io
+import shutil
+import tarfile
+import tempfile
+import sys
+from pprint import pformat
+
+from snakemake.executors import ClusterExecutor, sleep
+from snakemake.exceptions import WorkflowError
+from snakemake.logging import logger
+from snakemake.common import get_container_image, get_file_hash, async_lock
+from snakemake.remote.AzBlob import AzureStorageHelper
```

Comment:
Please move this import into the method where it is used. Otherwise, AzBlob will be loaded even if somebody uses nothing from azure, essentially turning the required libs a necessary dependency.